### DISCUSSION TITLE
MacBook Air M1 Big Sur 11.2 brew update error

### DISCUSSION THREAD

**Discussion by laoer** [Homebrew/discussions#673] [Everyday usage] [answered]
My MBA M1 update to Big Sur 11.2, run brew update, show error : 

brew update
==> Downloading https://homebrew.bintray.com/bottles/gettext-0.21.arm64_big_sur.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/339b62b52ba86dfa73091d37341104b46c01ae354ca425000732df68
######################################################################## 100.0%
==> Downloading https://homebrew.bintray.com/bottles/pcre2-10.36.arm64_big_sur.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/8160558f322198cb735708ca993a683594d6f9dc83112cc26a378be4
######################################################################## 100.0%
==> Downloading https://homebrew.bintray.com/bottles/git-2.30.0.arm64_big_sur.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/7e0f30621e1a5be89bf604eacdca8de9a79bd16c273a1d6f4dd39c6f
######################################################################## 100.0%
==> Installing dependencies for git: gettext and pcre2
==> Installing git dependency: gettext
==> Pouring gettext-0.21.arm64_big_sur.bottle.tar.gz
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/envsubst
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/gettext
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgattrib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgattrib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgattrib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgattrib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcat
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcat
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcat
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcat
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcmp
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcmp
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcmp
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcmp
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcomm
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcomm
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcomm
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgcomm
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgconv
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgconv
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgconv
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgconv
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgen
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgen
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgen
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgen
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgexec
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgexec
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgexec
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgexec
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgfilter
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgfilter
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgfilter
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgfilter
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgfmt
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgfmt
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgfmt
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgfmt
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msggrep
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msggrep
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msggrep
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msggrep
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msginit
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msginit
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msginit
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msginit
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgmerge
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgmerge
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgmerge
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgmerge
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgunfmt
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgunfmt
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgunfmt
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msgunfmt
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msguniq
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msguniq
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msguniq
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/msguniq
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/ngettext
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/recode-sr-latin
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/recode-sr-latin
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/xgettext
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/xgettext
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/xgettext
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/bin/xgettext
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/gettext/cldr-plurals
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/gettext/cldr-plurals
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/gettext/cldr-plurals
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/gettext/cldr-plurals
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/gettext/hostname
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/gettext/hostname
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/gettext/urlget
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/gettext/urlget
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/libasprintf.0.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/libgettextlib-0.21.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/libgettextlib-0.21.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/libgettextpo.0.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/libgettextpo.0.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/libgettextsrc-0.21.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/libgettextsrc-0.21.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/libgettextsrc-0.21.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/libgettextsrc-0.21.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/libintl.8.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/gettext/0.21/lib/libtextstyle.0.dylib
🍺  /opt/homebrew/Cellar/gettext/0.21: 1,953 files, 20.8MB
==> Installing git dependency: pcre2
==> Pouring pcre2-10.36.arm64_big_sur.bottle.tar.gz
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/pcre2/10.36/bin/pcre2grep
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/pcre2/10.36/bin/pcre2test
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/pcre2/10.36/bin/pcre2test
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/pcre2/10.36/bin/pcre2test
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/pcre2/10.36/bin/pcre2test
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/pcre2/10.36/lib/libpcre2-16.0.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/pcre2/10.36/lib/libpcre2-32.0.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/pcre2/10.36/lib/libpcre2-8.0.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/pcre2/10.36/lib/libpcre2-posix.2.dylib
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/pcre2/10.36/lib/libpcre2-posix.2.dylib
🍺  /opt/homebrew/Cellar/pcre2/10.36: 230 files, 5MB
==> Installing git
==> Pouring git-2.30.0.arm64_big_sur.bottle.tar.gz
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/bin/git
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/bin/git
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/bin/git-shell
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/bin/git-shell
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-daemon
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-daemon
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-http-backend
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-http-backend
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-http-fetch
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-http-fetch
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-http-push
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-http-push
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-imap-send
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-imap-send
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-remote-http
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-remote-http
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-sh-i18n--envsubst
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/libexec/git-core/git-sh-i18n--envsubst
==> Caveats
The Tcl/Tk GUIs (e.g. gitk, git-gui) are now in the `git-gui` formula.

zsh completions and functions have been installed to:
  /opt/homebrew/share/zsh/site-functions

Emacs Lisp files have been installed to:
  /opt/homebrew/share/emacs/site-lisp/git
==> Summary
🍺  /opt/homebrew/Cellar/git/2.30.0: 1,486 files, 41MB
==> `brew cleanup` has not been run in 30 days, running now...
Removing: /Users/tim/Library/Logs/Homebrew/nvm... (99B)
Pruned 1 symbolic links and 2 directories from /opt/homebrew
==> Caveats
==> git
The Tcl/Tk GUIs (e.g. gitk, git-gui) are now in the `git-gui` formula.

zsh completions and functions have been installed to:
  /opt/homebrew/share/zsh/site-functions

Emacs Lisp files have been installed to:
  /opt/homebrew/share/emacs/site-lisp/git
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18:  5325 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18:  5457 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18:  5538 Killed: 9               "$GIT_EXECUTABLE" "$@"
Error: Fetching /opt/homebrew failed!
Fetching /opt/homebrew/Library/Taps/homebrew/homebrew-core failed!
Fetching /opt/homebrew/Library/Taps/homebrew/homebrew-services failed!
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18:  5619 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18:  5717 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18:  5718 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18:  5736 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18:  5737 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18:  5755 Killed: 9               "$GIT_EXECUTABLE" "$@"
/opt/homebrew/Library/Homebrew/cmd/update.sh: line 18:  5756 Killed: 9               "$GIT_EXECUTABLE" "$@"
==> Homebrew has enabled anonymous aggregate formula and cask analytics.
Read the analytics documentation (and how to opt-out) here:
  https://docs.brew.sh/Analytics
No analytics have been recorded yet (nor will be during this `brew` run).

Error: Failure while executing; `git config --replace-all homebrew.analyticsmessage true` was terminated by uncaught signal KILL.

Has anyone have the same problem?

**Comment by laoer** [comment, marked as answer]
I found the problem, after upgrade to Big Sur 11.2, need reinstall Command Line Tools.
xcode-select --install

**Reply by sortega306** [reply]
For me it doesn't help. It seems that Git 2.30.0 ist broken. (codesign Error?)

**Reply by carlocab** [reply]
What does `brew config` and `brew doctor` say?

**Reply by arturasmckwcz** [reply]
Worked

**Reply by makarim178** [reply]
It worked for me as well, thank you

**Comment by sortega306** [comment]
> I finally was able to get this to work by brew reinstall gettext and pcre2 (the two dependencies of git).
https://github.com/Homebrew/discussions/discussions/439#discussioncomment-261267

Thats my solution :-)

**Reply by nowtp** [reply]
This worked for me as well, reinstalled gettext; pcre2 and then git

**Reply by sixone1** [reply]
Thats my solution too. Thanks.

**Reply by eddiehex** [reply]
it is fixed!!

**Comment by taylortechdigital** [comment]
Same issues will sigkill when simply calling "git" 

"xcode-select --install" worked for me!

**Comment by Yukimune** [comment]
```
xcode-select --install
brew uninstall git
(opt)rm -fr /opt/homebrew/etc/gitconfig
brew uninstall gettext pcre2
brew cleanup -s git gettext pcre2 
brew cleanup --prune-prefix
brew install git
```
In my case, the brew's cache caused.

**Reply by JolluSK** [reply]
This works, Thanks bro

**Reply by ALexanderSpiridonov** [reply]
yes, it is alive now! Thanks, man! I have spent a week trying to solve the git issue.

**Reply by RohanKumarSachdeva** [reply]
thank you so much, worked for me as well

**Reply by xulli1996** [reply]
thanks its work for me

**Reply by Ac-Srikanth** [reply]
Thanks a lot . Worked for me .

**Reply by simoncrypta** [reply]
Your system is ready to brew. 😁👍
Thank you for your solution!
( @Yukimune I just notice a typo in  "brew **uinstall** git" )

**Reply by DanRosenfeld** [reply]
Thanks so much @Yukimune! This was the only thing that worked for me as well. Much appreciated.

**Reply by kbnewlon** [reply]
Worked for me! Thank you!

**Comment by smallufo** [comment]
After upgrade Big Sur (M1 chip) from 11.2 to  11.2.1 , git failed again.
```
brew reinstall git 

==> Reinstalling git 
==> Pouring git-2.30.0.arm64_big_sur.bottle.tar.gz
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/bin/git
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/bin/git
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/bin/git-shell
Error: Failed applying an ad-hoc signature to /opt/homebrew/Cellar/git/2.30.0/bin/git-shell
...
```

And , after `xcode-select --install` , it works. (but I don't want xcode ...  )

**Reply by nowtp** [reply]
Correct me if I am wrong; but my understanding was that you needed that portion of Xcode for brew to work?

Plus the Xcode-select is just the command line tools and not the full Xcode install.

**Reply by smallufo** [reply]
yes , but in `11.2` , I can `brew install git` , without installing Xcode-select.

Today I updated to `11.2.1` , and git was broken. Even reinstall doesn't work. So I have to install `XCode-select` , which brings tons of luggages  under `/Library/Developer/CommandLineTools`  (just feel sucks)

**Reply by SMillerDev** [reply]
Brew will always need commandline tools indeed. `xcode-select --install` installs these (or selects them if you already have XCode)

**Reply by smallufo** [reply]
Oops. So it seems `xcode-select` was transferred from my previous machine. Sorry I didn't notice this. And today I need to reinstall `xcode-select` and everything seems working now.

**Reply by nowtp** [reply]
The only thing that I currently do not understand; is why with each update so far on Big Sur, command line tools breaks which then breaks git. 

Unless this (my limited understanding) is due to the sudo exploit patching?

**Reply by SMillerDev** [reply]
No, apple just seems to be removing it on updates. Not sure why.

**Comment by itsmelion** [comment]
My problem seemed to be slightly different.
```
Error: Couldn't find manifest matching bottle checksum.
/usr/local/Homebrew/Library/Homebrew/software_spec.rb:402:in `tab_attributes'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2387:in `bottle_tab_attributes'
...
```

But i managed to fix just by nuking the cache folder
`rm -rf $HOME/Library/Caches/Homebrew/downloads/`
with a fresh download no checksum mismatch was gone.
